### PR TITLE
feat: Add Flyout positioning and offset options

### PR DESCRIPTION
**Pull request by imoutaharik**

### CODE REVIEW

**Comment by thisguyscode** [comment]
@imoutaharik this little PR is going to make some people very happy, well done! Just one small comment from me.

**Comment by imoutaharik** [comment]
@thisguyscode Thanks, but where is the comment? I can't see it. ;) suspense.

File: src/components/Flyout/Flyout.tsx [hunk 1]
```diff
@@ -27,6 +27,13 @@ import { Overlay } from "./Overlay";
 export const FLYOUT_DIVIDER_COLOR = "#eaebeb";
 export const FLYOUT_DIVIDER_HEIGHT = "10px";
 
+export enum FlyoutPlacement {
+    Top = "top left",
+    Bottom = "bottom left",
+    Right = "right",
+    Left = "left",
```
Comment (jonasbuechel) [inline src/components/Flyout/Flyout.tsx:34]: Is aligning left for Top and Bottom the thing we like to enforce? Do we have a need for right-aligned as well?

File: src/components/Flyout/hooks/useOverlayPositionWithBottomMargin.ts [hunk 2]
```diff
@@ -14,25 +15,31 @@ type UseOverlayPositionWithBottomMarginProps = {
...
     overlayRef,
     scrollRef,
     isOpen,
+    placement,
+    offset,
 }: UseOverlayPositionWithBottomMarginProps) => {
     const overlayHeight = getTotalOverlayHeight(overlayRef, scrollRef);
 
     const isFlipped = shouldDisplayAbove(triggerRef, overlayHeight, FLYOUT_OVERLAY_OFFSET, INTERCOM_BUTTON_HEIGHT);
+    const verticalPosition = isFlipped ? FlyoutPlacement.Top : FlyoutPlacement.Bottom;
```
Comment (jonasbuechel) [inline src/components/Flyout/hooks/useOverlayPositionWithBottomMargin.ts:33]: If `placement = FlyoutPlacement.Top` and `isFlipped = false` it evaluates to Bottom.
The expected behavior would be displaying on top.

Screenshot 2022-05-23 at 11 29 01

**Review by jonasbuechel — CHANGES_REQUESTED** [review]
I added a story to easier see placements and found an issue when using "top left"

File: src/components/Flyout/Flyout.tsx (hunk 1 above)
Comment (imoutaharik) [reply src/components/Flyout/Flyout.tsx]: Top left and bottom left were originally enforced but I added the right position for more positioning flexibility.

File: src/components/Flyout/hooks/useOverlayPositionWithBottomMargin.ts (hunk 2 above)
Comment (imoutaharik) [reply src/components/Flyout/hooks/useOverlayPositionWithBottomMargin.ts]: I fixed it but the flip logic is handled manually as not all the flyout height is within the scroll area. It's calculated for top and bottom but we don't have flip logic done yet for right and left.

**Pull request merged**